topology: add TGL tplg for max98357a amp and ALC5682 Headset codec

sathya-nujella opened the pull request:
Add TGL topology for the DUT which has the following audio configuration:
        Max98357A speaker (I2S)
        ALC5682 headset codec (I2S)
        DMIC
        4 HDMI devices

sathya-nujella commented:
Hi @ranj063,
Removed line spacing and updated the patch. Please review.

lyakh commented on tools/topology/sof-tgl-max98357a-rt5682.m4:
```diff
@@ -0,0 +1,202 @@
+#
+# PCM0 <---> volume <----> SSP1  (Speaker - max98357a)
+# PCM1 <---> volume <----> SSP0  (Headset - ALC5682)
+# PCM99 ----> volume -----> DMIC01 (dmic0 capture)
```
I think arrows should be pointing in the opposite direction

sathya-nujella replied:
fixed and updated patch now.

lyakh commented on tools/topology/sof-tgl-max98357a-rt5682.m4:
```diff
@@ -0,0 +1,202 @@
+#
+# Define the pipelines
+#
+# PCM0 <---> volume <----> SSP1  (Speaker - max98357a)
```
this seems to be playback only, not duplex, so arrows one direction, please

sathya-nujella replied:
fixed and updated patch now.

sathya-nujella commented:
> comments inline

Thank you.
fixed and updated patch now.
Also fixed check-patch warnings.

sathya-nujella commented:
> For me looks ok, but I'd wait for @plbossart or @ranj063 review, to have a look from a driver side.

Hi @ranj063 , @plbossart 
Once this BUG: https://github.com/thesofproject/sof/issues/2084 is fixed.. we will update if any changes needed for this m4 file.  Also if we have to avoid any M/N settings after fix lands, we can add one more patch for this m4 file.
For now, this is the working m4 file on DUT. so, asking for your review of this one.

ranj063 commented:
@sathya-nujella could you please aliign the dmic Dec name with PR 2186?

sathya-nujella commented:
> @sathya-nujella could you please aliign the dmic Dec name with PR 2186?

Changed PCM name now and updated patch.

Also cross verified in DUT, reflected the change.
**Before:**
localhost /home/root # arecord -l
..
card 0: sofsofrt5682 [sof-sof_rt5682], device 99: **DMIC01** (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**After:**
localhost ~ # arecord -l
..
card 0: sofsofrt5682 [sof-sof_rt5682], device 99: **DMIC48kHz** (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

jajanusz commented:
@ranj063 is it ok now?
